report/ReportFiles.py

```diff
@@ -326,6 +327,50 @@ def write_histogram_image(out_prefix, annotations, filt_annot, index_rep):
     _, max_y = plt.ylim()
     plt.xlim((0, xm + 1))
 
+    # ----- PLOTLY HISTOGRAM -----
+    dist_text, dist_filt_text = [], []
```

Comment:
This block should be implemented with more "functional" approach - more readable and error prone (no indexing, clearer structure). Something like (untested):  
`def get_filt_text(dist_filt, dist):
    if dist_filt == 0: 
        return ""
    elif dist_filt != 0 and dist == 0:
        return str(dist_filt))        
    elif dist_filt != 0 and dist != 0:
        if dist_filt > dist:
            return str(dist_filt - dist)
        elif dist_filt == dist:
            return ""
        else:
            return str(dist_filt)
            
def get_dist_text(dist):
    if dist == 0: 
        return ""
    else:
        return str(dist)
        
dist_text = [get_dist_text(d) for d in dist]        // or even without function: dist_text = ["" if d == 0 else str(dist) for d in dist]
dist_text_filt = [get_filt_text(df, d) for df, d in zip(dist_filt, dist)]`

Reply:
Note: list.append() is very slow and the code with it is not very readable. It should be avoided in most uses. Dante was programmed when we weren't aware of it, so there is a lot of it lying around - let's not add more where possible.

Reply:
Replaced and tested label creation code block with example proposed above.